fuzz/fuzz_targets/fuzz_skrifa_outline.rs

```diff
@@ -154,11 +129,99 @@ fn do_glyf_things(outline_request: OutlineRequest, data: &[u8]) -> Result<(), Bo
     Ok(())
 }
 
+fn hinting_modes(hinted: bool) -> Vec<HintingMode> {
+    if !hinted {
+        return vec![HintingMode::default()];
+    }
+    vec![
+        HintingMode::Strong,
+        HintingMode::Smooth {
+            lcd_subpixel: None,
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: None,
+            preserve_linear_metrics: false,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Horizontal),
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Horizontal),
+            preserve_linear_metrics: false,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Vertical),
+            preserve_linear_metrics: true,
+        },
+        HintingMode::Smooth {
+            lcd_subpixel: Some(LcdLayout::Vertical),
+            preserve_linear_metrics: false,
+        },
+    ]
+}
+
 fuzz_target!(|data: &[u8]| {
-    let mut unstructured = Unstructured::new(data);
-    let Ok(outline_request) = unstructured.arbitrary() else {
+    // data from corpus is likely to be a font. If we chope off the head to make an outline request
+    // it is likely data is no longer a font. So, take the cross product of likely values for various options
+    // If a lot of values are possible choose randomly with rng seeded from data to ensure reproducible results.
+    let Ok(font) = FontRef::new(data) else {
         return;
     };
-    let data = unstructured.take_rest();
-    let _ = do_glyf_things(outline_request, data);
+    let mut requests = vec![OutlineRequest::default()];
+    requests = requests
+        .into_iter()
+        .flat_map(|r| {
+            fuzz_sizes()
+                .into_iter()
+                .map(move |size| OutlineRequest { size, ..r.clone() })
+        })
```

Comment:
I find this all very hard to follow, but I think at the very least you could start with,
```suggestion
    let requests = fuzz_sizes()
        .into_iter().map(move |size| OutlineRequest { size, ..Default::default() })
    })
```

and save a few lines?


it would also be a bit easier to read if it was in its own little fn.

Reply:
Migrated to fn with a comment explaining wht's happening